node/src/components/block_synchronizer/block_builder.rs

```diff
@@ -325,7 +329,7 @@ impl BlockBuilder {
                     self.promote_peer(maybe_peer);
                 }
                 Ok(FinalitySignatureAcceptance::Noop) => {
-                    // noop
+                    self.touch();
```

Comment:
we may not need to touch here after all, given the other fix we made later in need_next itself to check the inflight latch earlier

Reply:
I'll check again once your fixes are merged.